tasks/create-pyxis-image/create-pyxis-image.yaml

```diff
@@ -117,17 +117,24 @@ spec:
             DIGEST="${CONTAINER_IMAGE##*@}"
             PULLSPEC="docker://${REPOSITORY}@${DIGEST}"
             MEDIA_TYPE=$(skopeo inspect --raw "$PULLSPEC" | jq -r .mediaType)
-            skopeo inspect --no-tags "$PULLSPEC" > /tmp/skopeo-inspect.json
+            ARCHS=$(get-image-architectures "$PULLSPEC")
+            for ARCH in $ARCHS; do
+              IMAGE_ARCH=$(skopeo inspect --override-os linux --override-arch "$ARCH" \
+                           --format '{{.Digest}}' "$PULLSPEC")
+              ARCH_IDENTIFIED_PULLSPEC="${PULLSPEC%@*}@${DIGEST}"
+              skopeo inspect --no-tags "$ARCH_IDENTIFIED_PULLSPEC" > /tmp/skopeo-inspect-${ARCH}.json
 
-            PYXIS_CERT_PATH=/tmp/crt PYXIS_KEY_PATH=/tmp/key create_container_image \
-              --pyxis-url $PYXIS_URL \
-              --certified $(params.certified) \
-              --tags "$TAGS" \
-              --is-latest $(params.isLatest) \
-              --verbose \
-              --skopeo-result /tmp/skopeo-inspect.json \
-              --media-type "$MEDIA_TYPE" \
-              --rh-push $(params.rhPush) | tee /tmp/output
-
-            grep 'The image id is' /tmp/output | awk '{print $NF}' >> $(results.containerImageIDs.path)
+              PYXIS_CERT_PATH=/tmp/crt PYXIS_KEY_PATH=/tmp/key create_container_image \
+                --pyxis-url $PYXIS_URL \
+                --certified $(params.certified) \
+                --tags "$TAGS" \
+                --is-latest $(params.isLatest) \
+                --verbose \
+                --skopeo-result "/tmp/skopeo-inspect-${ARCH}.json" \
+                --media-type "$MEDIA_TYPE" \
+                --rh-push $(params.rhPush) | tee "/tmp/output-${ARCH}.json"
```

Comment:
You call the output file a json, but it's not a json. This is the output of the `create_container_image` script. I'm not sure exactly what's in there, but definitely some logging, including the `The image id is` line which we need to parse to get the image id.

And then the resulting json should contain a mapping like:
```json
{
    "components": [
        {
            "imageUrl": "image-url",
            "pyxisImages": [
                {
                    "arch": "amd64",
                    "imageID": "abcd1234"
                },
                {
                    "arch": "s390x",
                    "imageID": "abcd1236"
                }
            ]
        }
    ]
}
```

Reply:
Yeah. I was wondering that as well, having  it  for each component. In this initial test I only considered a single one just to provoke these good reviews.

Reply:
done